internal/store/segment/server.go

```diff
@@ -412,13 +412,27 @@ func (s *server) runHeartbeat(_ context.Context) error {
 					LeaderId: info.leader.Uint64(),
 					Term:     info.term,
 				}
-				if _, err := s.cc.ReportSegmentLeader(context.Background(), req); err != nil {
+				segment, err := s.cc.ReportSegmentLeader(context.Background(), req)
+				if err != nil {
 					log.Debug(ctx, "Report segment leader to controller failed.", map[string]interface{}{
 						"leader":     info.leader,
 						"term":       info.term,
 						log.KeyError: err,
 					})
 				}
+
+				if segment.State != "freezing" {
+					break
+				}
+
+				s.replicas.Range(func(key, value interface{}) bool {
+					b, _ := value.(replica)
+					if b.appender.Status().Leader.Equals(vanus.ID(segment.LeaderBlockId)) {
```

Comment:
use `LeaderBlockId` is unsafe, because it may be not equal to `info.leader` from `info.term`.